llarp/service/endpoint.cpp

```diff
@@ -254,6 +260,17 @@ namespace llarp
       }
 #endif
 
+      // deregister dead sessions
+      {
+        auto itr = m_DeadSessions.begin();
+        while(itr != m_DeadSessions.end())
+        {
+          if(itr->second->IsDone(now) && itr->second->ShouldRemove())
+            itr = m_DeadSessions.erase(itr);
+          else
+            ++itr;
+        }
```

Comment:
you could replace this block with:

```c++
std::erase(std::remove_if(m_DeadSessions.begin(), m_DeadSessions.end(), [now](const auto& val) {
  return (val.second->IsDone(now) && val.second->ShouldRemove());
}), m_DeadSessions.end());
```

Reply:
yea i could, should i?

Reply:
up to you. I prefer the STL version, but I'm a C++ weenie.